### ISSUE TITLE
Style/RedundantSelf false positive with Pathname#mkdir refinement

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop-hq/rubocop#8994]
## Expected behavior

I would expect/wish Rubocop wouldn't flag `self` as redundant when `self` is *not* an instance of `Pathname` in this case but an `Integer` of `0` instead.

## Actual behavior

The following is a code snippet where `super` is messaged and `self` is returned because `super` will answer an `Integer` so an explicit call to `self` is necessary in order to allow `#mkdir` to be chained for this refinement. Example:

![image](https://user-images.githubusercontent.com/50245/98191562-b09f5a00-1ed6-11eb-94d4-70f7b28f3e44.png)

More examples of this [Refinement](https://www.alchemists.io/projects/refinements/#_mkdir) are available if it helps to illustrate this refinement further.

## Steps to reproduce the problem

Save the following to `example.rb`:

``` ruby
refine Pathname do
  def mkdir
    self.exist? ? self : super and self
  end
end
```

Then run `rubocop example.rb` to see the *Style/RedundantSelf* error.

## RuboCop version

```
$ [bundle exec] rubocop -V
0.93.1 (using Parser 2.7.2.0, rubocop-ast 1.1.0, running on ruby 2.7.2 x86_64-darwin19)
```

**Comment by bbatsov** [comment]
I guess it should be possible to filter out definitions in refinements. Nobody was using them when the cop was conceived, so we didn't fret on them much back then. :-)

**Comment by marcandre** [comment]
Isn't the error about the `self.exist?`? The cop wants you to write `exist?` instead

**Comment by marcandre** [comment]
Also, I don't believe the following to be correct:

> when `self` is _not_ an instance of `Pathname` in this case but an `Integer` of `0` instead.

In that example, `self` *is always* a `Pathname`.

**Comment by bkuhlmann** [comment]
@bbatsov No worries. In this case, it wasn't due to being a refinement. 😅 

@marcandre Thanks, you are correct and sorry for not catching this sooner. Also, you are correct that `self` is not `0`. I was confusing that with the `0` that is answered back when `#mkdir` creates a new directory.
